### ISSUE TITLE
A few bugs and glitches while using the Timeline and Character options [1.4.1]

### ISSUE THREAD

**Issue by FermiBrudja** [coppolaemilio/dialogic#855]
My desire to cooperate to polish the plugin increased today, surprisingly (yay!) I'm an expert at breaking everything I install. I might be useful for once!

I tried to follow [the tutorial](https://www.youtube.com/watch?v=sYjgDIgD7AY) with a freshly installed 1.4.1 but I'm getting errors in the **Debugger**. Namely, after I changed the scale from 100% to 25% for the Character's portrait I started to get this error and the **Timeline** will not run beyond the first character join animation:
```
E 0:00:01.135   resume: Resumed function 'event_handler()' after yield, but script is gone. At script: res://addons/dialogic/Nodes/DialogNode.gd:636
  <C++ Error>   Method failed. Returning: Variant()
  <C++ Source>  modules/gdscript/gdscript_function.cpp:1795 @ resume()
```

And other glitches like, when you make a character join for the first time, if you don't toggle off the **Wait for animation**, the dialog box never appears and the **Timeline** stops. I get this debugger error:
```
E 0:00:02.648   resume: Resumed function 'event_handler()' after yield, but script is gone. At script: res://addons/dialogic/Nodes/DialogNode.gd:687
  <C++ Error>   Method failed. Returning: Variant()
  <C++ Source>  modules/gdscript/gdscript_function.cpp:1795 @ resume()
  <Stack Trace> DialogicAnimaTween.gd:82 @ finished_once()
```

As for some other oddities:

- The modified scale for the portraits never changes upon testing the game (I point to the first bug above)
- When another character talks after the first one, their name is colored White but it gets its color back after the next line of dialog appears. If they are just exchanging one-liners, the names are consistently kept white as long as you ping-pong dialogues between the characters.

It would be good to add a fade-in for the box dialog instead of appearing in a split second (I know, it's a suggestion but I managed to break my **Timeline** again trying to get rid of the dialog box before the character appears on scene). To fulfill my desire, I tried to force a **Wait** the length of the join animation of the first character but it didn't work as intented I got this error without hope to trigger the dialog box after the character joins with his animation:
```
E 0:00:02.641   resume: Resumed function 'event_handler()' after yield, but script is gone. At script: res://addons/dialogic/Nodes/DialogNode.gd:884
  <C++ Error>   Method failed. Returning: Variant()
  <C++ Source>  modules/gdscript/gdscript_function.cpp:1795 @ resume()
```

I'm guessing that the most recent update of Godot might have broken a few things in the plugin because it is not working as intended. My version is this if it helps in anything:
![image](https://user-images.githubusercontent.com/68450307/160492072-9e445edb-be39-4327-8785-97d67ba1ac28.png)

In a nutshell, the Timeline gets stuck upon the first character join and I'm forced to toggle off the Wait for Animation to make it to work. Changing the Scale of the portraits also stops the Timeline's flow after the first Character Join (only work-around I see is to scale the portraits manually in Photoshop to get the perfect fit) I forgot to say that the offset never manages to happen since the scale never gets applied. Positioning the characters' portraits is almost impossible from within the plugin's options 🤔

Hope it helps a bit because I really want this plugin to shine and use it extensively for my project which is massive in narrative and Dialogic would help tons to ease my workflow, if I just can make it beyond these odd happenings 😭

Thanks for reading /

**Comment by qskousen** [comment]
How are you adding the dialog?
I am running the same Godot version as you and installed this plugin last night (also same version), have enlarged (rather than shrunk) a character portrait, and had no trouble adding a timeline via script and playing it when triggered.

**Comment by FermiBrudja** [comment]
> 

Well, I followed the guide of Emilio (though a tad out-of-date but still useful regardless). Because I already have some experience with other visual novel apps, I went "a bit" out of the way and started by first adding a Background Image, then adding a Background Music and basically trying to copy the same layout as my other test project on a different app. 
From that just added the Join Character, but because in the original test the character came from the left of the screen, I added the Back In Left with a Length of 2... milliseconds? (I don't know which measure is used in that box) and left the box of "Wait for Animation" because I didn't want the text-box to appear before the character entered the cottage.

I added then his first Text (2 lines) then a different Text to change the expression of the face to follow the mood of the dialogue, after that an Audio Event for the apparition of the ghost which would Join the scene with a Join in a Fade In fashion and mirrored (because the ghost is on the right side of the screen).
After that, Character Update to use a "fear" expression, followed by his Text expressing surprise, then the ghost starts to talk with his own text a bit and then the scene despawns as shown by Emilio in the video.

Thing is, after trying it out I saw the character started obviously way too big because the original file is a full body of 750x2248 px and both the knight and the ghost needed to be scaled down. Before that is when I started to try to troubleshoot why the scene wasn't advancing beyond the Join of the first character, that is when I noticed that box of "Wait for Animation" and after toggling it OFF, I tested the scene and the scene rand without interruptions, but with the dialog box already on the scene before the character entered the cottage (without the text yet). I checked the box again and ran the scene to record whatever error it gave the Debugger, as noted on my first post.

I left the checkbox off for the first character and proceeded to scale down the models and after doing that it again got stuck with the error I posted. I returned the scale of both portraits to their defaults and the scene ran as normal (but I could only see the feet of the characters talking because of the SIZE).

And I did nothing else because I found very odd that my Dialogic was working like this. As I said, I am an expert at breaking stuff when they fall in my hands. I followed the guide of drag'n'drop the contents of the plugin and all of that so I don't think it's an installation error which puzzles me because I did nothing offensive to the plugin nor the project itself but to activate the plugin from its settings and nothing else 🤔 I did indeed added a few folders to hold the portrait graphics, the sound and the music but that is the extend of all the stuff I did during this past weekend, just adding these few assets to try them out with Dialogic.

I'm yet stumpted because it's happening to me and nobody else. Today I'll make a new project and redo all the steps with the same 2 characters, the same background image, music and SFX. If I get again this error I'll update, and if everything goes right... then guess I'll need to ditch that project and carry on with the new one 🧠

**Comment by FermiBrudja** [comment]
> Sorry, I should have been more clear. When I asked how you are adding the dialog, I meant how are you adding it to the scene - are you doing it via script, or adding the dialog scene into your scene via the Godot UI?

Ah no, like the video, everything from the Timeline panel. No single script was touched during the edition of the project, even after my curiosity drove me to the thought, I stayed my hand and steered away from opening any script because those things are too fragile.

**Comment by qskousen** [comment]
That is interesting. If you still have trouble after rebuilding it, a screenshot of the node tree on the left hand side might be helpful in debugging. Alternatively, if possible, you could upload the whole project somewhere so others can look at it and see if they have the same issue, and try to debug. So far I've been unable to reproduce your issue.

**Comment by AnidemDex** [comment]
For some reason, your dialog node dissappears after trying to handle the fade-in animation:
https://github.com/coppolaemilio/dialogic/blob/239feb7d2b673ad99969c3e000070416f07178ae/addons/dialogic/Nodes/DialogNode.gd#L632-L636

That could explain why:
> if you don't toggle off the Wait for animation, the dialog box never appears and the Timeline stops

I can't find on a direct solution

**Comment by zaknafean** [comment]
I think this ended up being fixed by some changes i made in 1.5 concerning the anima animation tweening, that would often break things if certain animation combinations were used. Going to call it closed for the time being.
